mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -78,31 +78,39 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
     return Value();
   }
 
-  Value backtraceToAlloc(Value inp) const {
-    if (auto bt = inp.getDefiningOp<miopen::TransformOp>())
-      return backtraceToAlloc(bt->getOperand(0));
-    if (auto bt = inp.getDefiningOp<memref::ExpandShapeOp>())
-      return backtraceToAlloc(bt->getOperand(0));
-    if (auto bt = inp.getDefiningOp<memref::AllocOp>())
-      return inp;
+  typedef SmallVector<miopen::TransformOp, 4> TransformList;
+
+  Value bwTraceTo(Value inp, TransformList *transforms = nullptr) const {
```

Comment:
This is `untransform()`

Reply:
with some extra bits

Reply:
Update: it has somewhat different semantics and so can probably stay its own thing